src/api/types.ts

```diff
@@ -62,21 +65,82 @@ export type Filter = 'latest' | 'pending' | FilterObject | void
 
 export interface ERC20Interface {
   address: Account,
-  getTotalSupply(): Promise<BigNumber>,
-  balanceOf(account?: Account): Promise<BigNumber>,
-  transfer(to: Account, value: Balance, sender: Account, tx?: TransactionObject): Promise<Receipt>,
-  transferFrom(sender: Account, to: Account, value: Balance, tx?: TransactionObject): Promise<Receipt>,
-  approve(spender: Account, value: Balance, sender: Account, tx?: TransactionObject): Promise<Receipt>,
+  totalSupply(): Promise<BigNumber>,
+  balanceOf(account: Account): Promise<BigNumber>,
+  transfer(to: Account, value: Balance, tx?: TransactionObject): Promise<Receipt>,
```

Comment:
IMO - as these only return a `bool` based on transfer success or not, this, `transferFrom && approve` should have a manadatory `tx` param

Reply:
Yes, you're right